cmd/node_problem_detector.go

```diff
@@ -69,32 +50,41 @@ func main() {
 
 	npdo.ValidOrDie()
 
-	monitors := make(map[string]types.Monitor)
+	// Initialize problem daemons.
+	problemDaemons := problemdaemon.NewProblemDaemons(npdo.MonitorConfigPaths)
+
+	// Legacy per-monitor configuration.
+	// This section is kept so that existing command line option still works on NPD.
+	// TODO(xueweiz): when NPD's refactoring is finished, remove legacy configuration.
 	for _, config := range npdo.SystemLogMonitorConfigPaths {
-		if _, ok := monitors[config]; ok {
```

Comment:
I think we still need to keep this check? For the case: `-- system-log-monitor=config1,config1,config2`. Notice `config1` is specified twice.

Reply:
Good point. I just merged the two sections of initialization code together. `problemdaemon.NewProblemDaemons()` is handling this logic now.